azext_hanaonazure/custom.py

```diff
@@ -67,3 +67,24 @@ def enable_monitoring_hanainstance(client, resource_group_name, instance_name, h
         "hanaDbPassword": hana_db_password
     }
     return client.enable_monitoring(resource_group_name, instance_name, monitoring_details)
+
+def list_sapmonitor(client):
+    return client.list()
+
+def show_sapmonitor(client, resource_group_name, monitor_name):
+    return client.get(resource_group_name, monitor_name)
+
+def create_sapmonitor(client, resource_group_name, monitor_name, region, hana_subnet, hana_hostname, hana_db_sql_port, hana_db_username, hana_db_password, hana_db_name):
+    monitoring_details = {
+        "location": region,
```

Comment:
What is the purpose of this location? what do we need to do if this location is different than where the HanaDB is?

Reply:
I don't think this would be 100% related to the HanaDB, but to the Vnet in which the monitoring compute resource is going to be deployed.

And I guess this "location" would indicate which RP to call.

Reply:
Ya, this is the location (region) of where you want to SAP Monitor to be created in.
Before, we used the same region as the HANA Instance (baremetal blade), but since we are not created SAP Monitor on a specific blade, the location needs to be passed in.